dev/cuda/Makefile

```diff
@@ -8,8 +8,20 @@ ifeq ($(NVCC),)
 		$(error nvcc not found.)
 endif
 
+ifneq ($(CI),true) # if not in CI, then use the GPU query
+  ifndef GPU_COMPUTE_CAPABILITY # set to defaults if: make GPU_COMPUTE_CAPABILITY= 
+    GPU_COMPUTE_CAPABILITY = $(shell __nvcc_device_query) # assume if NVCC is present, then this likely is too
+    GPU_COMPUTE_CAPABILITY := $(strip $(GPU_COMPUTE_CAPABILITY))
+  endif
+endif
+
 # Compiler flags
-CFLAGS = -O3 --use_fast_math
+ifeq ($(GPU_COMPUTE_CAPABILITY),) # set to defaults if: make GPU_COMPUTE_CAPABILITY= 
+  CFLAGS = -O3 --use_fast_math 
+else
+  CFLAGS = -O3 --use_fast_math --generate-code arch=compute_$(GPU_COMPUTE_CAPABILITY),code=[compute_$(GPU_COMPUTE_CAPABILITY),sm_$(GPU_COMPUTE_CAPABILITY)]
```

Comment:
this makes it impossible to generate code for more than one arch at a time, which might be useful, e.g., for sass export, or if you  want to run the same executable on  A100 and H100 nodes  without recompiling (and having to remember to recompile, if it's a shared cluster file system)

Reply:
Got it - at one point this was a request but it's easy to remove.

Reply:
it is a good convenience option, just want to make sure we're aware of its limitations; especially when we want  to build sass in CI for both  A100 and H100

Reply:
I see - should we add an explicit step to generate sass for the A100 and H100?

Reply:
Added a change for this.